### PR TITLE
[Team H (2023)] Added wiki entry for running ROS on an unsupported OS

### PR DESCRIPTION
**Pull request by sohambhave**

### CODE REVIEW

File: wiki/common-platforms/ros/ros-unsupported-os.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+
+## Setups
+### 1. Running ROS2 Humble via a docker container on Ubuntu 20.04
+
+#### Installing Docker
+Follow instructions given here to install docker - https://roboticsknowledgebase.com/wiki/tools/docker/
+#### Running ROS2 Humble docker container
+Running the command below will pull the latest ROS2 Humble Docker Image, start a container and attach a shell to it
+```
```
Comment (nevalsar) [inline wiki/common-platforms/ros/ros-unsupported-os.md:38]: Specify language for nicer, syntax-highlighted snippets.
```suggestion
```sh
```

File: wiki/common-platforms/ros/ros-unsupported-os.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+ros2 run demo_nodes_cpp talker
+```
+```
+ros2 run demo_nodes_py listener
+```
+
+#### Creating a custom docker image using ROS2 Humble
+We can also create a custom docker image from ROS2 Humble base image by adding the following snippet to the Dockerfile:
+```
```
Comment (nevalsar) [inline wiki/common-platforms/ros/ros-unsupported-os.md:52]:
```suggestion
```Dockerfile
```

File: wiki/common-platforms/ros/ros-unsupported-os.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+#### Installing Docker
+Follow instructions given here to install docker - https://roboticsknowledgebase.com/wiki/tools/docker/
+#### Running ROS2 Humble docker container
+Running the command below will pull the latest ROS2 Humble Docker Image, start a container and attach a shell to it
+```
+docker run -it ros:humble bash
+```
+
+By running the following commands in two separate terminals, you should see the talker saying that it’s Publishing messages and the listener saying I heard those messages. This verifies both the C++ and Python APIs are working properly.
```
Comment (nevalsar) [inline wiki/common-platforms/ros/ros-unsupported-os.md:42]: The instructions are a little hard to follow here. After running the above command, there's only one terminal that's attached to the running container's pseudo-TTY, but the below steps require two separate terminal sessions. Please update instructions to include naming the container during creation and connecting to the existing container from multiple terminals.

File: wiki/common-platforms/ros/ros-unsupported-os.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+## See Also:
+- https://roboticsknowledgebase.com/wiki/tools/docker/
+
+## Further Reading
+- https://docs.docker.com/engine/reference/run/
+
+## References
+- https://docs.ros.org/en/humble/Installation/Alternatives.html
+- https://github.com/osrf/docker_images
+- https://hub.docker.com/_/ros/
```
Comment (nevalsar) [inline wiki/common-platforms/ros/ros-unsupported-os.md:184]: 'See Also' and 'Further Reading' are redundant sections and can be merged.

For each link below, please add a descriptive title and link using hyperlinks instead of raw URLs.

**Review by nevalsar — CHANGES_REQUESTED** [review]
Largely a well-written page but needs some changes.

Please make the necessary changes and re-raise the review request.

File: _data/navigation.yml [hunk 5]
```diff
@@ -65,6 +65,8 @@ wiki:
         url: /wiki/common-platforms/ros/ros-navigation/
       - title: ROS Global Planner
         url: /wiki/common-platforms/ros/ros-global-planner/
+      - title: Running ROS on unsupported OS
```
[the remark this answers is not in the document]
Comment (sohambhave) [reply _data/navigation.yml]: Added this change

File: wiki/common-platforms/ros/ros-unsupported-os.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+# overwritten except in special circumstances. 
+# You should set the date the article was last updated like this:
+date: 2023-05-03 # YYYY-MM-DD
+# This will be displayed at the bottom of the article
+# You should set the article's title:
+title: How to run a ROS distribution on an operating system that doesn't natively support it?
+# The 'title' is automatically displayed at the top of the page
+# and used in other parts of the site.
+---
+Each ROS distribution supports only a select number of platforms, and for all supported platforms, readily available Debian packages are available which can be easily installed using package managers such as apt. However, there often arise situations where you need to install and use a ROS distribution on a platform that is not supported for the distribution. This issue is a common occurrence when attempting to run specific ROS distributions on Single Board Computers (SBCs). The problem often arises due to the SBC's published images being either too outdated or too recent to support the required ROS distribution. This article will describe how to install and run ROS distributions in such cases.
```
[the remark this answers is not in the document]
Comment (sohambhave) [reply wiki/common-platforms/ros/ros-unsupported-os.md]: Added this change

File: wiki/common-platforms/ros/ros-unsupported-os.md (hunk 1 above)
Comment (sohambhave) [reply wiki/common-platforms/ros/ros-unsupported-os.md]: Added this change

File: wiki/common-platforms/ros/ros-unsupported-os.md (hunk 2 above)
Comment (sohambhave) [reply wiki/common-platforms/ros/ros-unsupported-os.md]: Added this change

File: wiki/common-platforms/ros/ros-unsupported-os.md (hunk 3 above)
Comment (sohambhave) [reply wiki/common-platforms/ros/ros-unsupported-os.md]: Added this change

File: wiki/common-platforms/ros/ros-unsupported-os.md (hunk 4 above)
Comment (sohambhave) [reply wiki/common-platforms/ros/ros-unsupported-os.md]: Added this change

**Comment by sohambhave** [comment]
Have made all the changes. Please re-review it.

File: wiki/common-platforms/ros/ros-unsupported-os.md (hunk 3 above)
Comment (nevalsar) [reply wiki/common-platforms/ros/ros-unsupported-os.md]: Thank you!

**Review by nevalsar — APPROVED** [review]
Thank you for resolving the comments! Approving.
